Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -383,20 +390,18 @@ extension SwiftLanguageServer {
     _ = handle
   }
 
-  func rewriteCompletionPlacholders(_ completion: String) -> String {
-    if !completion.contains("<#") {
-      return completion
-    }
-
-    var result = completion
+  func rewriteSourceKitPlaceholders(inString string: String, clientSupportsSnippets: Bool) -> String {
+    var result = string
     var index = 1
     while let start = result.range(of: "<#") {
       guard let end = result[start.upperBound...].range(of: "#>") else {
-        log("invalid placholder in \(completion)", level: .debug)
-        return completion
+        log("invalid placeholder in \(string)", level: .debug)
+        return string
       }
       // FIXME: add name to placeholder
-      result.replaceSubrange(start.lowerBound..<end.upperBound, with: "${\(index):value}")
+      let displayName = "value"
+      let placeholder = clientSupportsSnippets ? "${\(index):\(displayName)}" : displayName
```

Comment:
I would have expected that we would omit the string entirely.  Otherwise you have to manually delete it before you can type in the new value.

Reply:
Good point, keeping the names will be useful for the method names in refactoring, but we can add support for that once they actually support snippets.